core/src/main/java/org/mskcc/cbio/portal/scripts/ImportResourceData.java

```diff
@@ -0,0 +1,618 @@
+package org.mskcc.cbio.portal.scripts;
+
+import org.mskcc.cbio.portal.dao.*;
+import org.mskcc.cbio.portal.model.*;
+import org.mskcc.cbio.portal.util.*;
+
+import java.io.*;
+import joptsimple.*;
+import java.util.*;
+import java.util.regex.*;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections.map.MultiKeyMap;
+
+public class ImportResourceData extends ConsoleRunnable {
+
+    public static final String DELIMITER = "\t";
+    public static final String METADATA_PREFIX = "#";
+    public static final String SAMPLE_ID_COLUMN_NAME = "SAMPLE_ID";
+    public static final String PATIENT_ID_COLUMN_NAME = "PATIENT_ID";
+    public static final String RESOURCE_ID_COLUMN_NAME = "RESOURCE_ID";
+    public static final String URL_COLUMN_NAME = "URL";
+    public static final String SAMPLE_TYPE_COLUMN_NAME = "SAMPLE_TYPE";
+    private int numSampleSpecificResourcesAdded = 0;
+    private int numPatientSpecificResourcesAdded = 0;
+    private int numStudySpecificResourcesAdded = 0;
+    private int numEmptyResourcesSkipped = 0;
+    private int numSamplesProcessed = 0;
+
+    private static Properties properties;
+
+    private File resourceDataFile;
+    private CancerStudy cancerStudy;
+    private ResourceTypes resourceType;
+    private boolean relaxed;
+    private Set<String> patientIds = new HashSet<String>();
+
+    public static enum MissingResourceValues {
+        NOT_APPLICABLE("Not Applicable"), NOT_AVAILABLE("Not Available"), PENDING("Pending"),
+        DISCREPANCY("Discrepancy"), COMPLETED("Completed"), NULL("null"), MISSING(""), NA("NA");
```

Comment:
are these missing values different from some other enum already in the code? This seems like something that's not resource specific

Reply:
No, I borrow the same definition from the `Clinical Data`. I can import it from that class also.

Reply:
yeah there can maybe be some shared class with common N/A values. I do think some of these don't really look like N/A to me, e.g. `Discrepancy`, `Pending` and `Completed`. That seems very specific to some pipeline that we have set up

Reply:
@inodb I haven't found common null values in the project, so I created one `ValuesUtils.java`